App/ActionExtension/ActionToBraveViewController.swift

```diff
@@ -0,0 +1,118 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import MobileCoreServices
+import UniformTypeIdentifiers
+
+class ActionToBraveViewController: UIViewController {
+  
+  private enum SchemeType {
+    case url, query
+  }
+
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+      
+    // Get the item[s] for handling from the extension context
+    for item in extensionContext?.inputItems as? [NSExtensionItem] ?? [] {
+      for provider in item.attachments ?? [] {
+        
+        // Opening browser with search url
+        if provider.hasItemConformingToTypeIdentifier(UTType.text.identifier) {
+          loadAttachmentFor(type: .query, using: provider)
+          break
+        }
+    
+        // Opening browser with site
+        if provider.hasItemConformingToTypeIdentifier(UTType.url.identifier) {
+          loadAttachmentFor(type: .url, using: provider)
+          break
+        }
+      }
+    }
+  }
+  
+  func done() {
+    // Return any edited content to the host app, in this case empty
+    extensionContext!.completeRequest(returningItems: [], completionHandler: nil)
```

Comment:
do we have to force unwrap here?

Reply:
No certainly not, good catch. 

Fixed at https://github.com/brave/brave-ios/pull/8682/commits/640b3481565a6f54ab6bf9830aa289d1a26d5d68